### PR TITLE
Makes flashlights use cells

### PR DESCRIPTION
**Pull request by Yoshax**
- Flashlights now use cells
- These can be swapped
- Flashlights are now marginally brighter across the board
- Flashlights now have a selectable brightness level
- Only flashlights, maglights, and lanterns use this
- Flashlights will spawn with a random amount of power between 25% and 100%
- A full flashlight on low will last 66 minutes, one on medium will last 33 minutes, one on low will last 16 minutes. Numbers are open to change and feedback.
- Charge amount is obfuscated and shown as a general description such as, "Average" rather than a full percent when examining the flashlight

Aim is to add more atmosphere and flavor. It would be more fun, and exciting if you are in maintenance looking for something, a changeling, traitor, etc and you lose track of your charge and it runs out.

I realise the low duration is not very realistic, but the aim of it is fun, flavor and atmosphere over realism.

### CODE REVIEW

**Comment by Datraen** [comment]
Travis is being a git about formatting comments.

**Comment by Yoshax** [comment]
Ya, I got it. Typo'd a ' as a ;

**Comment by Techhead0** [comment]
Things I noticed:
- `attackby()` should probably end in `else ..()`
- Flares and lamps will spawn with cells (which can be removed and replaced)
- Penlights are noticeably smaller than their cells and might end up exploited as cell storage devices

**Comment by Yoshax** [comment]
Resolved, resolved and resolved.

**Comment by Anewbe** [comment]
I like the concept a great deal, but I have the following concerns:

1. I would like if they used the device cells, rather than the full sized batteries. Feels more believably sized, and it'll keep the times on a refill consistent.

2. Do the items with power_use = 0 still have the ability to switch brightness levels? Strikes me as an oversight if so, because why would anyone ever use anything else if there's no downside?

3. I don't think the initial charge should be randomized. I'd rather not find myself preparing for a blackout, then having to immediately charge a flashlight to make sure it's actually full.

Also might be wise to throw a thread on the forums about this, people might have opinions.

**Comment by Yoshax** [comment]
1. I disagree, to do so you would have to make the device cells far more prevalent and easy to acquire. This would upset the balance of your energy weapons greatly, or it would make acquiring cells for the lights too difficult. Power cells are fairly common, engineering have a lot, tool storage always has a lot, and there's other easy places to get them.

2. They do not, the verb is removed within the parent new() if they do not have power_use != 0

3. That's part of the fun of it. It's rather dull that everything on station is perfect and pristine, it really adds to the feel that many dislike that every shift is super new and stuff. The randomisation could be knocked up a bit to between say 50% and 100%, but I would like it to some degree to add more flavor and a more 'lived-in' feel to things.

**Comment by zaredman** [comment]
If you want guaranteed light in an 'oh shit' situation, grab a flare.

**Comment by Yoshax** [comment]
Or a glowstick.

**Comment by redstryker** [comment]
Yeah, the glowsticks are there for a reason. :P  They burn twice as long as flares.

**Comment by Anewbe** [comment]
Alright then, I'm fine with leaving it as a normal power-cell if you remove the randomization.

**Comment by Yoshax** [comment]
I've noted in the OP that flashlights are margiinally brighter across the case, I thought I had made this change but it seems I never did. It also means my last commit message is slightly incorrect. Which will be corrected in a moment.

**Comment by Yoshax** [comment]
Should be good now.

File: html/changelogs/Yoshax-Flashlights.yml [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+# Any changes you've made.  See valid prefix list above.
+# INDENT WITH TWO SPACES.  NOT TABS.  SPACES.
+# SCREW THIS UP AND IT WON'T WORK.
+# Also, all entries are changed into a single [] after a master changelog generation. Just remove the brackets when you add new entries.
+# Please surround your changes in  double quotes ("), as certain characters otherwise screws up compiling. The quotes will not show up in the changelog.
+changes: 
+  - rscadd: "Flashlights now take powercells."
+  - rscadd: "Flashlights now have multiple brightness levels, including low, medium and high."
+  - rscadd: "The cells that spawn in flashlights will spawn with a random amount of charge between 25% and 100%"
```
Comment (Anewbe) [inline html/changelogs/Yoshax-Flashlights.yml]: Looks like this was changed, yes?

**Pull request merged**